sixMans/game.py

```diff
@@ -377,28 +379,45 @@ async def process_self_picking_teams(self, emoji, user, added=True):
             await self.update_game_info()
             await self._notify(Strings.ONGOING_GS)
 
+    async def _remove_stale_reactions(self, emoji_hex: int, member: discord.Member):
+        if self.reaction_lock:
+            return
+
+        self.reaction_lock = True
+        coros = []
+
+        for react_hex, reaction in [(react_hex, self._get_pick_reaction(react_hex)) for react_hex in SELECTION_MODES]:
+            if react_hex != emoji_hex:
+                coros.append(self.info_message.remove_reaction(reaction, member))
+
+        await asyncio.gather(*coros)
+        self.reaction_lock = False
+
     async def process_team_select_vote(self, emoji, member, added=True):
         if member not in self.players:
+            asyncio.create_task(self.info_message.remove_reaction(emoji, member))
             return
 
-        if self._hex_i_from_emoji(emoji) not in SELECTION_MODES:
+        emoji_hex = self._hex_i_from_emoji(emoji)
+
+        if emoji_hex not in SELECTION_MODES:
             return
 
-        # RECORD VOTES
-        votes = {}
-        self.info_message = await self.textChannel.fetch_message(self.info_message.id) # this is needed to get the up to date reactions for a message
-        for this_react in self.info_message.reactions:
-            this_react:discord.Reaction
-            react_hex_i = self._hex_i_from_emoji(this_react.emoji)
-            if react_hex_i in SELECTION_MODES:
-                reacted_members = await this_react.users().flatten()
-                reacted_players = [player for player in reacted_members if player in self.players]  # Intersection of reacted_members and self.players
-                if added and this_react.emoji != emoji and member in reacted_players:
-                    await this_react.remove(member)
-                    reacted_players.remove(member)
-                votes[react_hex_i] = len(reacted_players)
+        if added:
+            await self._remove_stale_reactions(emoji_hex, member)
+            self.player_votes[member] = emoji_hex
+        elif self.player_votes[member] == emoji_hex:
+            self.player_votes.pop(member)
+
+        # ensure we still need to take action
+        if self.reaction_lock or self.teamSelection.lower() != Strings.VOTE_TS.lower():
+            return
+
```

Comment:
Lots of extra empty lines in here. I don't mind them to separate different function groups, etc. But generally avoid within functions. I think this could be condensed to 1 empty line, or a comment could be added on one.